### ISSUE TITLE
Single-species carbon simulations still not working for CO, CO2, and OCS in GCHP

### ISSUE THREAD

**Issue by msulprizio** [geoschem/geos-chem#2237]
### Name and Institution (Required)

Name: Melissa Sulprizio
Institution: Harvard

### Description of your issue or question

Even after the merge of #1992, the single species carbon simulations for CO, CO2, and OCS are still failing in GCHP. This was revealed by adding these simulations to integration tests in #2236. The issue is in the run directory creation for these simulations. Wildcards do not work in HISTORY.rc for GCHP so each species must be listed under the diagnostic collection's fields. For carbon, this looks like:

```
#==============================================================================
  SpeciesConc.template:       '%y4%m2%d2_%h2%n2z.nc4',
  SpeciesConc.format:         'CFIO',
  SpeciesConc.timestampStart: .true.
  SpeciesConc.monthly:        1
  SpeciesConc.frequency:      010000
  SpeciesConc.duration:       240000
  SpeciesConc.mode:           'time-averaged'
  SpeciesConc.fields:         'SpeciesConcVV_CH4',  'GCHPchem',
                              'SpeciesConcVV_CO ',  'GCHPchem',
                              'SpeciesConcVV_CO2',  'GCHPchem',
                              'SpeciesConcVV_OCS',  'GCHPchem',
```

The code in singleCarbonSpecies.sh updates HISTORY.rc to remove the carbon species that aren't needed.

https://github.com/geoschem/geos-chem/blob/64796d8f399a3e539694e47baccce00ff01fc735/run/shared/singleCarbonSpecies.sh#L218-L225

However, this results in the removal of the line

```
  SpeciesConc.fields:         'SpeciesConcVV_CH4',  'GCHPchem',
```

for CO, CO2, and OCS simulations and therefore no fields defined for the SpeciesConc collection. For example, the resulting HISTORY.rc for CO looks like:

```
#==============================================================================
  SpeciesConc.template:       '%y4%m2%d2_%h2%n2z.nc4',
  SpeciesConc.format:         'CFIO',
  SpeciesConc.timestampStart: .true.
  SpeciesConc.monthly:        1
  SpeciesConc.frequency:      010000
  SpeciesConc.duration:       240000
  SpeciesConc.mode:           'time-averaged'
                              'SpeciesConcVV_CO ',  'GCHPchem',
                              ##'SpeciesConcMND_CO ', 'GCHPchem',
::
```

In addition, for CO2 simulations instances of `_CO2` and `EmisCO2` are also removed from the configuration files because they include "CO". The resulting SpeciesConc collection in HISTORY.rc for CO2 looks like:

```
#==============================================================================
  SpeciesConc.template:       '%y4%m2%d2_%h2%n2z.nc4',
  SpeciesConc.format:         'CFIO',
  SpeciesConc.timestampStart: .true.
  SpeciesConc.monthly:        1
  SpeciesConc.frequency:      010000
  SpeciesConc.duration:       240000
  SpeciesConc.mode:           'time-averaged'
::
```

**Comment by yantosca** [comment]
@msulprizio: Thanks for letting me know about this.  I am testing a fix:
```bash
function updateHistory() {

    #========================================================================
    # Removes entries in HISTORY.rc for unused species
    #
    # Arguments:
    # ${1} : List of species to exclude
    # ${2} : Path to the run directory
    #========================================================================

    # File to be modified
    file="${2}/HISTORY.rc"

    # For GCHP: remove entries for species to be excluded
    for spc in ${1}; do
	sed -i "/\_${spc}'/d"   "${file}"
        sed -i "/\_${spc} '/d"  "${file}"
        sed -i "/Emis${spc}_/d"  "${file}"
    done

    # Also disable emissions for OCS-only simulations
    # (as we currently do not have any)
    isItemInList "OCS" "${1}"
    if [[ $? == 1 ]]; then
	sed -i -e "s/'Emissions/#'Emissions/" "${file}"
    fi
}```

**Comment by yantosca** [comment]
On further review, the issue is caused by the `.fields` tag being deleted when removing `HISTORY.rc` entries for CH4.  by the `singleCarbonSpecies.sh`  script.  This left the e.g. Emissions collection definition looking like this:
```console
  Emissions.template:         '%y4%m2%d2_%h2%n2z.nc4',
  Emissions.format:           'CFIO',
  Emissions.timestampStart:   .true.
  Emissions.monthly:          0
  Emissions.frequency:        010000
  Emissions.duration:         010000
  Emissions.mode:             'time-averaged'
                              'EmisCO2_Total        ', 'GCHPchem',
                              'EmisCO2_FossilFuel   ', 'GCHPchem',
                              'EmisCO2_Ocean        ', 'GCHPchem',
                              'EmisCO2_BalBiosph    ', 'GCHPchem',
                              'EmisCO2_NetTerrExch  ', 'GCHPchem',
                              'EmisCO2_Ship         ', 'GCHPchem',
                              'EmisCO2_Aviation     ', 'GCHPchem',
                              'EmisCO2_CO2SurfCorr  ', 'GCHPchem',
                              'EmisCO2_BiomassBurn  ', 'GCHPchem',
```
We will have to put in some more logic to work around this.

@lizziel: Do you know if the line containing the `.fields` tag has to be populated?  Could it be a comment?

**Comment by yantosca** [comment]
The best solution might be to have different HISTORY.rc files for the single-species GCHP carbon simulations, and to have the `createRunDir.sh` script place the relevant file in the run directory.

**Comment by msulprizio** [comment]
This has been resolved by #2236.
